docs/routing.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Blazor routing
+author: guardrex
+description: Learn how to route requests in a client-side Blazor app and about the NavLink component.
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 05/01/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: client-side/blazor/routing
+---
+# Blazor routing
+
+By [Luke Latham](https://github.com/guardrex)
+
+[!INCLUDE[](~/includes/blazor-preview-notice.md)]
+
+Learn how to route requests in a client-side Blazor app and about the NavLink component.
+
+[View or download sample code](https://github.com/aspnet/Blazor.Docs/tree/master/docs/components/common/samples/) ([how to download](xref:client-side/blazor/index#view-and-download-samples)). See the [Get started](xref:client-side/blazor/get-started) topic for prerequisites.
+
+## Route templates
+
+Routing in Blazor is achieved by providing a route template to each accessible component in the app.
+
+When a *\*.cshtml* file with an `@page` directive is compiled, the generated class is given a [RouteAttribute](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.mvc.routeattribute) specifying the route template. At runtime, the router looks for component classes with a `RouteAttribute` and renders whichever component has a route template that matches the requested URL.
+
+Multiple route templates can be applied to a component. In the [sample app](https://github.com/aspnet/Blazor.Docs/tree/master/docs/components/common/samples/), the following component responds to requests for `/BlazorRoute` and `/DifferentBlazorRoute`.
+
+*Pages/BlazorRoute.cshtml*:
+
+[!code-cshtml[](components/common/samples/2.x/ComponentsSample/Pages/BlazorRoute.cshtml?start=1&end=4)]
+
+## Route parameters
+
+Blazor components can receive route parameters from the route template provided in the `@page` directive. The Blazor client-side router uses route parameters to populate the corresponding component parameters.
```

Comment:
The Blazor client-side router uses route parameters to populate the corresponding component parameters with the same name (case insensitive).